src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -156,6 +159,76 @@ BinaryCompactObject::BinaryCompactObject(
         context,
         "Cannot have periodic boundary conditions with a binary domain");
   }
+
+  // Create block names and groups
+  static std::array<std::string, 6> wedges{"UpperZ", "LowerZ", "UpperY",
+                                           "LowerY", "UpperX", "LowerX"};
+  const auto add_object_region = [this](
+                                     const std::string& object_name,
+                                     const std::string& region_name) noexcept {
+    for (const std::string& wedge : wedges) {
+      // NOLINTNEXTLINE(performance-inefficient-string-concatenation)
+      const std::string block_name = object_name + region_name + wedge;
+      block_names_.push_back(block_name);
+      block_groups_[object_name + region_name].insert(block_name);
+    }
+  };
+  const auto add_object_interior =
+      [this](const std::string& object_name) noexcept {
+        const std::string block_name = object_name + "Interior";
+        block_names_.push_back(block_name);
+      };
+  const auto add_outer_region =
+      [this](const std::string& region_name) noexcept {
+        for (const std::string& wedge : wedges) {
+          for (const std::string& leftright : {"Left", "Right"}) {
+            if ((wedge == "UpperX" and leftright == "Left") or
```

Comment:
This block of code skips names like `OuterShellLeftUpperX` because they do not exist (this is good!), but still produces names like `OuterShellRightUpperX` (which implies the existence of non-existent blocks like `OuterShellRightLowerX`), when a better name (imo) would be `OuterShellUpperX` (perhaps also add a comment about how `Left` and `Right` are with respect to the y-z plane such that objects named `Left` are located to the -x direction of the y-z plane, and correspondingly for the `Right` objects.)